app/front/scripts/services/validation.js

```diff
@@ -1,47 +1,22 @@
 'use strict';
 
 var _ = require('lodash');
-var utils = require('../../../services/utils');
+var goodtables = require('goodtables');
 
-var goodTablesUrl = '//goodtables.okfnlabs.org/api/run';
+var goodtablesOptions = {
+  apiUrl: 'https://goodtables.io/api',
+  apiToken: 'D0123458B8E36326C60253FE4A7FF6662CAB0C48',
+  apiSourceId: '9b6b6391-5404-4e7f-bdb8-271c2cb42fbb'
+};
 
 angular.module('Application')
   .factory('ValidationService', [
-    '$q', 'Configuration',
-    function($q, Configuration) {
+    '$q',
+    function($q) {
       return {
         validateResource: function(source) {
-          var validationResult = {
-            state: 'checking'
-          };
-          if (typeof(source) !== 'string') {
-            validationResult.$promise = $q(function(resolve, reject) {
-              utils.validateData(source.data, undefined, undefined,
-                goodTablesUrl)
-                .then(resolve)
-                .catch(reject);
-            });
-          } else {
-            validationResult.$promise = $q(function(resolve, reject) {
-              utils.validateData(undefined, source, undefined, goodTablesUrl)
-                .then(resolve)
-                .catch(reject);
-            });
-          }
-          validationResult.$promise
-            .then(function(results) {
-              validationResult.state = 'completed';
-              if (results && results.errors && results.errors.length) {
-                validationResult.errors = results.errors;
-              }
-              return results;
-            })
-            .catch(function(error) {
-              validationResult.state = null;
-              Configuration.defaultErrorHandler(error);
-            });
-
-          return validationResult;
+          if (source.blob) source = source.blob;
```

Comment:
I think it's better to avoid multiple statements in a single line, and specially omitting the curly braces. It isn't dangerous here, but patterns have a tendency to spread. For an interesting story about bugs caused by single-line ifs, see https://medium.com/@jonathanabrams/single-line-if-statements-2565c62ff492.

Reply:
Nevermind what I said above. Although I personally prefer to always use braces, this is fine :+1: